lib/Bio/EnsEMBL/DataCheck/Checks/Population.pm

```diff
@@ -0,0 +1,75 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::Population;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'Population',
+  DESCRIPTION => 'Population data is consistent',
+  GROUPS      => ['variation_import'], 
+  DB_TYPES    => ['variation'],
+  TABLES      => ['population']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_length = 'Population size'; 
+  my $diag_length = 'Population has no stored size'; 
+  my $sql_length = qq/
+      SELECT p.population_id
+      FROM population p, sample_population sp
+      WHERE p.population_id = sp.population_id
```

Comment:
It's somewhat a matter of style, so feel free to ignore this one, but... I always find it easier to read a SQL query if the join conditions are separated out from filter conditions. For example, I would write this as:
`SELECT p.population_id`
`FROM population p `
`INNER JOIN sample_population sp USING (population_id)`
`WHERE p.size IS NULL`